### PR TITLE
[Execution] add execution scripts engine

### PR DESCRIPTION
**Pull request by zhangchiqing**
This PR moves the scripts execution functionality from the ingestion engine to a new script executor engine in order to reduce the complexity of the ingestion engine.

### CODE REVIEW

File: engine/execution/ingestion/engine.go [hunk 1]
```diff
@@ -1048,105 +1046,6 @@ func (e *Engine) matchAndFindMissingCollections(
 	return missingCollections, nil
 }
 
-func (e *Engine) ExecuteScriptAtBlockID(
```
Comment (zhangchiqing) [inline engine/execution/ingestion/engine.go:1051]: moved to scripts/engine.go

File: engine/execution/ingestion/engine_test.go [hunk 2]
```diff
@@ -1315,73 +1314,6 @@ func TestExecutionGenerationResultsAreChained(t *testing.T) {
 	execState.AssertExpectations(t)
 }
 
-func TestExecuteScriptAtBlockID(t *testing.T) {
```
Comment (zhangchiqing) [inline engine/execution/ingestion/engine_test.go:1318]: moved to scripts/engine_test.go

**Review by pattyshack — APPROVED** [review]

**Review by janezpodhostnik — APPROVED** [review]

**Comment by zhangchiqing** [comment]
bors merge

**Comment by zhangchiqing** [comment]
bors retry

**Comment by zhangchiqing** [comment]
bors retry

**Pull request merged**